testsuites/CBLTester/System_test_multiple_device/test_p2p_replication_eventing.py

```diff
@@ -0,0 +1,290 @@
+                                                        attachments_generator=attachment.generate_png_100_100)
+        server_doc_ids = db_obj_server.create_bulk_docs(num_of_docs, "server_doc", db=cbl_db_server,
+                                                        channels=channels,
+                                                        attachments_generator=attachment.generate_png_100_100)
+    else:
+        client_doc_ids = db_obj_client.create_bulk_docs(num_of_docs, "client_doc", db=cbl_db_client,
+                                                        channels=channels)
+        server_doc_ids = db_obj_server.create_bulk_docs(num_of_docs, "server_doc", db=cbl_db_server,
+                                                        channels=channels)
+
+    # Now set up client
+
+    repl = peer_to_peer_client.configure(host=server_host, server_db_name=db_name_server, client_database=cbl_db_client,
+                                         continuous=continuous, replication_type=replicator_type,
+                                         endPointType=endpoint_type)
+    repl_listener = peer_to_peer_client.addReplicatorEventChangeListener(repl)
+    peer_to_peer_client.client_start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    total = replicator.getTotal(repl)
+    completed = replicator.getCompleted(repl)
+    event_changes = peer_to_peer_client.getReplicatorEventChanges(repl_listener)
+    assert total == completed, "replication from client to server did not completed " + \
+                               str(total) + " not equal to " + str(completed)
+    server_docs_count = db_obj_server.getCount(cbl_db_server)
+    client_docs_count = db_obj_client.getCount(cbl_db_client)
+    assert server_docs_count == 2 * num_of_docs, "Number of docs is not equivalent to number of docs in Server "
+    assert client_docs_count == 2 * num_of_docs, "Number of docs is not equivalent to number of docs in Client "
+    replicator.stop(repl)
+    event_dict = get_event_changes(event_changes)
+    push_docs = []
+    pull_docs = []
+    error_docs = []
+    for doc in event_dict:
+        if event_dict[doc]['push'] is True:
+            push_docs.append(doc)
+        else:
+            pull_docs.append(doc)
+        if event_dict[doc]['error_code'] is not None:
+            error_docs.append({doc: "{}, {}".format(event_dict[doc]['error_code'],
+                                                    event_dict[doc]['error_domain'])})
+
+    # 5. Validating the event counts and verifying the push and pull event against doc_ids
+    assert sorted(push_docs) == sorted(
+        client_doc_ids), "Replication event push docs are not equal to expected no. of docs to be pushed"
+    assert sorted(pull_docs) == sorted(
+        server_doc_ids), "Replication event pull docs are not equal to expected no. of docs to be pulled"
+    assert len(error_docs) == 0, "Error found in replication events {}".format(error_docs)
+
+
+@pytest.mark.sanity
```

Comment:
you can remove sanity tag.  add replicating event tag to all tests

Reply:
Removed